src/ORM/Connect/DuplicateEntryException.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SilverStripe\ORM\Connect;
+
+/**
+ * Exception for errors related to duplicate entries (e.g. violating a unique index)
+ */
+class DuplicateEntryException extends DatabaseException
```

Comment:
New exception class so we can catch it in `DataObject::write()`.
Subclass of the existing `DatabaseException` so anyone already doing a `try {} catch (DatabaseException) {}`  will continue to catch this.